### PR TITLE
Add support for Quarkus CLI TLS command

### PR DESCRIPTION
**Pull request by michalvavrik**
### Summary

We need to test https://quarkus.io/guides/tls-registry-reference#quarkus-cli-commands-and-development-ca-certificate-authority so adding few convenient methods to facilitate testing.

Please check the relevant options

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Release (follows conventions described in the [RELEASE.md](https://github.com/quarkus-qe/quarkus-test-framework/blob/main/RELEASE.md))
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Example scenarios has been updated / added
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/AbstractCliCommand.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    public abstract AbstractCliCommand addToApplicationProperties(String... additions);
+
+    public File getApplicationProperties() {
+        var pathToSrcMainResources = "src" + File.separator + "main" + File.separator + "resources";
+        return app.getFileFromApplication(pathToSrcMainResources, "application.properties");
+    }
+
+    public String getApplicationPropertiesAsStr() {
```
Comment (rsvoboda) [inline quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/AbstractCliCommand.java:41]: Why not
```suggestion
    public String getApplicationPropertiesAsString() {
```
?

File: quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/QuarkusCliClient.java [hunk 2]
```diff
@@ -235,6 +235,8 @@ private Process runCli(Path workingDirectory, File logOutput, String... args) {
         cmd.addAll(Arrays.asList(args));
 
         if (QuarkusProperties.disableBuildAnalytics()) {
+            // TODO: if logic behind setting disabled analytics in our FW get revision
```
Comment (rsvoboda) [inline quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/QuarkusCliClient.java:238]: It would be good to have QQE to plan this effort.

**Review by rsvoboda — APPROVED** [review]
Approving, just 2 minor things that are not mandatory for merging.
I will wait till CI is done.

File: quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/QuarkusCliClient.java (hunk 2 above)
Comment (michalvavrik) [reply quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/QuarkusCliClient.java:238]: I'll create one and link it here little later. I need little break now.

File: quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/AbstractCliCommand.java (hunk 1 above)
Comment (michalvavrik) [reply quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/AbstractCliCommand.java:41]: Yes, I'll do that, pushing in a moment.

**Comment by michalvavrik** [comment]
> I will wait till CI is done.

Just FYI I run all related tests locally so I hope it will be fine. But sure, no merging till it's green.

File: quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/QuarkusCliClient.java (hunk 2 above)
Comment (michalvavrik) [reply quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/QuarkusCliClient.java:238]: I have created tracker https://issues.redhat.com/browse/QQE-935 and added it into this PR changes; I'll fill it's content when I am back from break.

File: quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/AbstractCliCommand.java (hunk 1 above)
Comment (michalvavrik) [reply quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/AbstractCliCommand.java]: Done.

File: quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/QuarkusCliClient.java (hunk 2 above)
Comment (michalvavrik) [reply quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/QuarkusCliClient.java:238]: see line 240

Comment (rsvoboda) [reply quarkus-test-cli/src/main/java/io/quarkus/test/bootstrap/QuarkusCliClient.java:238]: 🤦

**Comment by rsvoboda** [comment]
`Error: 6,695 INFO  [ERROR]   TlsCommandTest.testKeystoreInDefaultTlsRegistry:31 NullPointer Cannot invoke "java.security.KeyStore.aliases()" because "ks" is null`

```
 13:47:08,831 INFO  /home/runner/work/quarkus-test-framework/quarkus-test-framework/quarkus-dev-cli tls generate-certificate -c Dumbledore --name=dev-certificate --*** -d /home/runner/work/quarkus-test-framework/quarkus-test-framework/examples/quarkus-cli/target/quarkus-tls-command-tests/654ba4d1-604c-4402-99a9-05bcd15a67ff/tls-command-test
13:47:12,832 INFO  Command tls is not available, looking for available plugins ...
Error: 2,832 INFO  [ERROR] ❗  Command tls is missing and can't be installed.
Error: 2,832 INFO  [ERROR] ❗  Unmatched arguments from index 0: 'tls', 'generate-certificate', '-c', 'Dumbledore', '--name=dev-certificate', '--***', '-d', '/home/runner/work/quarkus-test-framework/quarkus-test-framework/examples/quarkus-cli/target/quarkus-tls-command-tests/654ba4d1-604c-4402-99a9-05bcd15a67ff/tls-command-test'
```

**Comment by rsvoboda** [comment]
`tls` is plugin of cli, maybe that's a problem?

**Comment by michalvavrik** [comment]
The cause is here: `Command tls is missing and can't be installed.` which I don't know why works locally and not in CI. I simply have to look what is different as I build CLI with same commands as CI does.

**Comment by michalvavrik** [comment]
> `tls` is plugin of cli, maybe that's a problem?

it needs Quarkus TLS Registry  extension, which is present though. I'll find the difference and write it here.

**Comment by michalvavrik** [comment]
I have reproduced it in brand new fork few times. Funny business, it seems that same command fails depending on parent directories of a Quarkus application created with CLI command. Then I deleted that fork and started a new fork and can't reproduce it again. There is probably some algorithm to it. I'll try to find reproducer.

**Comment by michalvavrik** [comment]
What I have changed is to perform test in temporary directory instead. I hope it's alright for now.

**Comment by michalvavrik** [comment]
Okay, I am thinking it is about some lock when the command is first executed during test because I cannot reproduce it outside of it.

**Comment by michalvavrik** [comment]
CI is green, merging.

**Pull request merged**